src/main/java/com/squareup/kotlinpoet/Util.kt

```diff
@@ -113,3 +113,58 @@ internal fun stringLiteralWithQuotes(value: String): String {
     return result.toString()
   }
 }
+
+internal fun isIdentifier(name: String): Boolean {
+  return IDENTIFIER_REGEX.matches(name)
+}
+
+internal fun isKeyword(name: String): Boolean {
+  return KEYWORDS.contains(name)
+}
+
+internal fun isName(name: String): Boolean {
+  return NAME_REGEX.split(name, 0).toTypedArray().none { isKeyword(name) }
```

Comment:
There's an overload of `split` that takes a `CharSequence` instead of a `Regex`. Can we just use that with `"."`?